### PR TITLE
ticdc/pipeline: sorter Node actor refactor

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve? 
this is a part of table pipeline actor model refactor #3881

### What is changed and how it works?
1 init the sorter with actor mode
2. set the WaitGroup 
3. add a new function to try to handle the pipeline message non-blocking

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/actor_node_context_test.go [hunk 1]
```diff
@@ -78,6 +78,16 @@ func TestThrow(t *testing.T) {
 	}
 }
 
+func TestActorNodeContextTrySendToNextNode(t *testing.T) {
+	ctx := NewContext(sdtContext.TODO(), nil, 1, &context.ChangefeedVars{ID: "zzz"}, &context.GlobalVars{})
```
Comment (overvenus) [inline cdc/processor/pipeline/actor_node_context_test.go:82]: Could you add `t.Parallel()`?

File: cdc/processor/pipeline/sorter.go [hunk 2]
```diff
@@ -267,16 +280,23 @@ func (n *sorterNode) Receive(ctx pipeline.NodeContext) error {
 					model.NewResolvedPolymorphicEvent(0, n.barrierTs))
 			}
 		}
+		if n.isTableActorMode {
+			return n.sorter.TryAddEntry(ctx, msg.PolymorphicEvent)
+		}
```
Comment (overvenus) [inline cdc/processor/pipeline/sorter.go:285]: Please add some comment and a TODO the remove the flag in future.

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: cdc/processor/pipeline/actor_node_context_test.go (hunk 1 above)
Comment (sdojjy) [reply cdc/processor/pipeline/actor_node_context_test.go:86]: done

File: cdc/processor/pipeline/sorter.go (hunk 2 above)
Comment (sdojjy) [reply cdc/processor/pipeline/sorter.go:287]: done

**Review by overvenus — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**